services/QuillLMS/app/models/classroom.rb

```diff
@@ -59,6 +59,10 @@ class Classroom < ApplicationRecord
 
   accepts_nested_attributes_for :classrooms_teachers
 
+  def destroy 
+    ClassroomsTeacher.where(classroom_id: id).destroy_all
```

Comment:
So how are students classrooms getting destroyed?

Reply:
Maybe, the class_name: "StudentsClassrooms" is giving the dependent: :destroy macro the information it needs to do the cascading deleting?

Reply:
I tried that already - unfortunately it does not affect the issue.

Reply:
That still begs the question, why doesn't `dependent: :destroy` work for teachers_classrooms but it does for students_classrooms

Reply:
I think you should add a comment / todo or something indicating the weirdness so that future developers don't go down this rabbit hole again with the `dependent: :destroy`

Reply:
Good idea - added.